proposals/Exceptions.md

```diff
@@ -170,21 +174,101 @@ also be sure to also pop off the caught exception values.
 
 ### Rethrows
 
-The _rethrow_ instruction has no arguments, and can only appear in catch
-blocks. It always re-throws the exception caught by the catch block. This allows
-the catch block to clean up state before the exception is passed back to the
-next enclosing try block.
+The _rethrow_ instruction can only appear in the body of a catch block.  The
+_rethrow_ instruction always re-throws the exception caught by the referenced
+catch block. This allows the catch block to clean up state before the exception
+is passed back to the next enclosing try block.
+
+The _rethrow_ instruction gets a _catch index_ argument. It is to disambiguate
```

Comment:
We could model this as an ordinary label, which must be the label of a `try` instruction. Then no new concept is needed.